src/Entity/MigrationProcessPipeline.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+
+/**
+ * Defines the Migration process pipeline entity.
+ *
+ * The migration process pipeline entity is a process plugin that can be
+ * exported to configuration and contains a suite of process plugins to run.
+ * It can be used in migrations as a single process plugin.
+ *
+ * @ConfigEntityType(
+ *   id = "oe_migration_process_pipeline",
+ *   label = @Translation("Migration Process Pipeline"),
+ *   module = "oe_migration",
+ *   handlers = {
+ *   },
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label"
+ *   },
+ *   config_export = {
+ *     "id",
+ *     "label",
+ *     "description",
+ *     "module",
+ *     "process",
+ *   },
+ * )
+ */
+class MigrationProcessPipeline extends ConfigEntityBase implements MigrationProcessPipelineInterface {
+
+  /**
+   * The migration process pipeline ID (machine name).
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The human-readable label for the migration process pipeline.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The configuration describing the process plugins.
+   *
+   * @var array
+   */
+  protected $process = [];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getProcess(array $tokens = []) {
+    return $this->getProcessNormalized($this->process, $tokens);
+  }
+
+  /**
+   * Resolve shorthands into a list of plugin configurations.
+   *
+   * @param array $process
+   *   A process configuration array.
+   * @param array $tokens
+   *   A process configuration array.
+   *
+   * @return array
+   *   The normalized process configuration.
+   */
+  protected function getProcessNormalized(array $process, array $tokens) {
```

Comment:
This method is not needed. You can move this logic directly in the main process definitions getter method.